mne_bids/sidecar_updates.py

```diff
@@ -0,0 +1,170 @@
+"""Update BIDS directory structures and sidecar files meta data."""
+# Authors: Adam Li <adam2392@gmail.com>
+#          Austin Hurst <mynameisaustinhurst@gmail.com>
+#          mne-bids developers
+# License: BSD (3-clause)
+
+import json
+from collections import OrderedDict
+
+from mne_bids.utils import _write_json
+
+
+# XXX: should we encode this, or let bids-validator do its job?
+# def _check_specification_compat(datatype):
```

Comment:
if we go this route I would add the schema files to the mne-bids repo without any alteration.

eg https://github.com/bids-standard/bids-validator/blob/master/bids-validator/validators/json/schemas/meg.json

it means we need a python way to validate a json schema

Reply:
> it means we need a python way to validate a json schema

https://python-jsonschema.readthedocs.io/en/stable/validate/

seems straightforward

Reply:
you're welcome: https://github.com/bids-standard/bids-validator/blob/master/bids-validator/bids_validator/bids_validator.py

Reply:
> you're welcome: https://github.com/bids-standard/bids-validator/blob/master/bids-validator/bids_validator/bids_validator.py

But this is just for filename verification… not for JSON file content?

Reply:
Seems like ideally this is handled in pybids, or bids-validator. I'll remove these.